fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -290,20 +289,62 @@ private void loadFromProto(@Nonnull RecordMetaDataProto.MetaData metaDataProto,
                                @Nonnull Descriptors.FileDescriptor[] dependencies,
                                boolean processExtensionOptions) {
         recordsDescriptor = buildFileDescriptor(metaDataProto.getRecords(), dependencies);
-        validateRecords(recordsDescriptor);
-        unionDescriptor = initRecordTypes(processExtensionOptions);
+        unionDescriptor = initRecordTypes(recordsDescriptor, processExtensionOptions);
         loadProtoExceptRecords(metaDataProto);
         processSchemaOptions(processExtensionOptions);
     }
 
     private void loadFromFileDescriptor(@Nonnull Descriptors.FileDescriptor fileDescriptor,
                                         boolean processExtensionOptions) {
         recordsDescriptor = fileDescriptor;
-        validateRecords(fileDescriptor);
-        unionDescriptor = initRecordTypes(processExtensionOptions);
+        unionDescriptor = initRecordTypes(fileDescriptor, processExtensionOptions);
         processSchemaOptions(processExtensionOptions);
     }
 
+    @Nonnull
+    private Descriptors.Descriptor initRecordTypes(@Nonnull Descriptors.FileDescriptor fileDescriptor, boolean processExtensionOptions) {
+        Descriptors.Descriptor unionDescriptor = fetchUnionDescriptor(fileDescriptor);
+        validateRecords(fileDescriptor, unionDescriptor);
+        fillUnionFields(unionDescriptor, processExtensionOptions);
+        return localUnionDescriptor == null ? unionDescriptor : localUnionDescriptor;
+    }
+
+    @Nonnull
+    private Descriptors.Descriptor fetchUnionDescriptor(@Nonnull Descriptors.FileDescriptor fileDescriptor) {
+        @Nullable Descriptors.Descriptor union = null;
```

Comment:
This means that you can't use an explicit usage on `DEFAULT_UNION_NAME` to use it some other way, which maybe is okay.

Reply:
That's right.